middle_end/flambda2/simplify/simplify_binary_primitive.ml

```diff
@@ -964,14 +964,26 @@ let simplify_array_load (array_kind : P.Array_kind.t)
     in
     (* CR mshinwell/vlaviron: if immutable array accesses were consistently
        setting [mutability] to [Immutable], we could restrict the following code
-       to immutable loads only and use [T.meet_is_immutable_array] instead. *)
-    match T.prove_is_immutable_array (DA.typing_env dacc) array_ty with
-    | Unknown -> contents_unknown ()
-    | Proved (elt_kind, fields, _mode) -> (
-      match elt_kind with
+       to immutable loads only and use [T.meet_is_array] instead. *)
```

Comment:
I think this comment needs to be updated. First, as discussed on Slack, we can use `T.meet_is_array` now, even for mutable loads.
The rest of the comment explains why it would be problematic to return `Invalid` on mismatched mutability (between the primitive and the type), so I propose to update it this way:
```ocaml
(* The frontend does not distinguish between mutable and immutable primitives yet,
   so we cannot assume that [mutability] must match the type of the array. *)
```